src/lib.rs

```diff
@@ -174,14 +174,31 @@ macro_rules! iproduct {
 /// # }
 /// ```
 macro_rules! izip {
-    ($I:expr) => (
-        (::std::iter::IntoIterator::into_iter($I))
-    );
-    ($($I:expr),*) => (
-        {
-            $crate::multizip(($(izip!($I)),*))
-        }
-    );
+    () => {
+        ::std::iter::empty::<()>()
+    };
```

Comment:
I'd prefer to not add this as a new feature. This iterator is better if we syntactically ensure there is at least one iterator.